kolibri/core/content/api.py

```diff
@@ -1316,6 +1319,75 @@ def consolidate(self, items, queryset):
         return sorted_items
 
 
+class ContentDownloadRequestViewset(ReadOnlyValuesViewset, CreateModelMixin):
+    serializer_class = serializers.ContentDownloadRequestSeralizer
+
+    pagination_class = OptionalPageNumberPagination
+
+    queryset = ContentDownloadRequest.objects.all()
+
+    values = (
+        "id",
+        "requested_at",
+        "reason",
+        "contentnode_id",
+        "metadata",
+        "status",
+        "facility",
+        "source_id",
+    )
+
+    def filter_queryset(self, queryset):
+        return queryset.filter(source_id=self.request.user.id)
+
+    def annotate_queryset(self, queryset):
+        return queryset.annotate(
+            has_removal=Exists(
+                ContentRemovalRequest.objects.filter(
+                    source_model=OuterRef("source_model"),
+                    source_id=OuterRef("source_id"),
+                    contentnode_id=OuterRef("contentnode_id"),
+                    requested_at__gte=OuterRef("requested_at"),
+                )
+            )
+        ).filter(has_removal=False)
+
```

Comment:
Still need to add a `delete` method here. Once you do that, you can completely remove the viewset below, as it is not needed.